Runtime/Triangulator.cs

```diff
@@ -2698,60 +2666,142 @@ private static void Eigen(double2x2 matrix, out double2 eigval, out double2x2 ei
         private static double2x2 Kron(double2 a, double2 b) => math.double2x2(a * b[0], a * b[1]);
     }
 
-    internal interface IUtils<T, T2> where T : unmanaged where T2 : unmanaged
+    internal readonly struct TranslationInt32 : ITransform<TranslationInt32, int, int2, long>
     {
-        T Const(float v);
+        public readonly TranslationInt32 Identity => new(int2.zero);
+        public readonly long AreaScalingFactor => 1;
+
+        private readonly int2 translation;
+
+        public TranslationInt32(int2 translation) => this.translation = translation;
+        private static TranslationInt32 Translate(int2 offset) => new(offset);
+        public static TranslationInt32 operator *(TranslationInt32 lhs, TranslationInt32 rhs) => new(
+            lhs.translation + rhs.translation
+        );
+
+        public TranslationInt32 Inverse() => new(-translation);
+        public int2 Transform(int2 point) => point + translation;
+        public long TransformArea(long area) => area * AreaScalingFactor;
+
+        public readonly TranslationInt32 CalculatePCATransformation(NativeArray<int2> positions)
+        {
+            throw new NotImplementedException("PCA is not implemented for integer coordinates.");
+        }
+
+        public readonly TranslationInt32 CalculateLocalTransformation(NativeArray<int2> positions)
+        {
+            int2 min = int.MaxValue, max = int.MinValue, com = 0;
+            foreach (var p in positions)
+            {
+                min = math.min(p, min);
+                max = math.max(p, max);
+                com += p;
+            }
+
+            com /= positions.Length;
+            // Note: For integer coordinates we only translate points to the origin, but do not scale them.
+            return Translate(-com);
+        }
+    }
+
+    /// <summary>
+    /// Helper interface for math operations that are generic over the coordinate type.
+    /// </summary>
+    /// <typeparam name="T">The raw coordinate type for a single axis. For example float or int.</typeparam>
+    /// <typeparam name="T2">The 2D coordinate composed of Ts. For example float2.</typeparam>
+    /// <typeparam name="TFloat">A floating point value. Used for angles and fractional values. For example float.</typeparam>
+    /// <typeparam name="TSquare">A squared coordinate. Used for squared distances and other products. For example float or long.</typeparam>
```

Comment:
Maybe not for you, but for anyone else reading the internals, having an explanation of what the 4 different generic types are used for is certainly useful. I don't get why you are so against even minimal amounts of code documentation?

Reply:
I believe this PR illustrates why writing comments for internal code can be problematic. The comments haven't been properly updated after recent changes, which can lead to confusion. 

Programmers do not use notepads in these days (and if they do, they are probably doomed). It is better to learn about generics from it usage, and an IDE is your best tool for managing these tasks! This type of minimalism aligns with the principles we promote in this project.

Reply:
It has been updated, just not pushed to github yet. I have a bunch of commits locally which update these comments, among other things.

> It is better to learn about generics from it usage, and an IDE is your best tool for managing these tasks

Sure, it will just take a person several times longer, compared to having actual documentation explaining things. An IDE will never explain *why* code looks a certain way.